CMake and GCC on Linux

lacc97 opened the pull request:
This PR adds a CMake build system (#7) and fixes some compilation issues with GCC on Linux.

I have also included small dependencies (stb, imgui, glad, tiny_obj_loader) as part of the source tree, and added appropriate CMake targets for linking. I'm somewhat unsure about this, but I really don't know how to otherwise include these dependencies into the CMake build system.

I haven't tested the CMake stuff on Windows.

lacc97 committed:
Fix missing includes

lacc97 committed:
Fix include HDR/HDRLoader.h

lacc97 committed:
Generalize force inlining for compilers besides VC

Add new Config.h file in root for compiler detection purposes.
Define new macro CFG_FORCE_INLINE for compiler-dependent force inlining attributes.

lacc97 committed:
Generalize access as char array of __m256i type

lacc97 committed:
Add CMake build system

Targets glad, imgui, stb and tiny_obj_loader are not defined or imported. They must be defined somewhere else.

Zielon commented:
Thx! Let me test it on both platforms.

Zielon reviewed (changes requested):
Please don't add files of 3rd party libraries. Instead use a [script](https://github.com/Zielon/CPURasterizer/blob/main/install.bat) with vcpkg but for linux. Sth like

```
mkdir Build
cd Build
git clone https://github.com/Microsoft/vcpkg.git vcpkg.linux
cd vcpkg.linux
git checkout 2020.11
./bootstrap-vcpkg.sh

./vcpkg install \
	glad:x64-linux \
	glfw3:x64-linux \
	glm:x64-linux \
	imgui:x64-linux \
	stb:x64-linux \
	tinyobjloader:x64-linux
```

lacc97 committed:
Use vcpkg dependencies

Add script to set up vcpkg
Split CMake/Dependencies.cmake file for easier dependency management in case user is not using vcpkg.

lacc97 commented:
Ok, added the script to setup vcpkg, and now CMake should integrate with and use the vcpkg dependencies. The only thing is that the user has to setup the VCPKG_ROOT environment variable before calling cmake so that the dependencies can actually be found and used.

Zielon commented:
Thank you. I will test it on both platforms and let you know.

Zielon commented:
Thank you for the contribution!
